### PR TITLE
Implement tests for range and partial content responses

### PR DESCRIPTION
**Pull request by vitoordaz**
Added tests for handling partial content responses and retries for incomplete response bodies.

# Which issue does this PR close?

I didn't file an issue, because this PR only adds unit tests.

# Rationale for this change
 
This improves code coverage and effectively documents how client behaves in edge case situation.

# What changes are included in this PR?

Only unit tests for GetClient

# Are there any user-facing changes?

No

### CODE REVIEW

File: src/client/get.rs [hunk 1]
```diff
@@ -579,6 +579,83 @@ mod http_tests {
...
+            .body("ello".to_string())
+            .unwrap();
+
+        mock.push(resp);
+
+        let b = store.get_range(&path, 1..5).await.unwrap();
+        assert_eq!(b.as_ref(), b"ello");
+
+        // NOTE: if debug_assertions is true, hyper panics with response content length header
+        // value does not match the length of response body.
```
Comment (alamb) [inline src/client/get.rs:597]: Does this mean hyper would panic in a real world case? If so, is there a ticket tracking it?

 Or does this mean that this particular response is not valid (and thus we shouldn't be testing it)?

**Review by alamb — COMMENTED** [review]
Thanks @vitoordaz  -- I took a quick look and it seems like a nice addition to me. I have one test question, otherwise I think this is ready to go

**Comment by alamb** [comment]
I think if you can merge up from main the CI tests would also pass cleanly

File: src/client/get.rs (hunk 1 above)
Comment (vitoordaz) [reply src/client/get.rs:597]: hyper panics in test environment which usually have `debug_assertions = true`. In prod `debug_assertions = false` and hyper returns regular response.

I'm wondering if we can update CI to run all tests with `debug_assertions = false` as well?

Comment (alamb) [reply src/client/get.rs:597]: > I'm wondering if we can update CI to run all tests with debug_assertions = false as well?

We could add a new test run with that enabled (or perhaps we could run the tests in `--release` mode)?

**Comment by alamb** [comment]
(please just merge up from main so we can get a clean CI run)

File: src/client/get.rs (hunk 1 above)
Comment (vitoordaz) [reply src/client/get.rs:597]: yes, I think that this will work. I was using this command to run this test

```
RUSTFLAGS="-C debug-assertions=false" RUST_BACKTRACE=full cargo test --features http client::get::http_tests::test_stream_retry
```

Comment (alamb) [reply src/client/get.rs:597]: Perhaps you can do so as a follow on PR
